client/cmd.go

```diff
@@ -147,6 +151,28 @@ func ReadPersistentCommandFlags(clientCtx Context, flagSet *pflag.FlagSet) (Cont
 		}
 	}
 
+	if clientCtx.GRPCClient == nil || flagSet.Changed(flags.FlagGRPC) {
```

Comment:
Can we please invert this condition and return early if .GRPCClient != nil or if !flagSet.Changed?

Reply:
But we don't actually want an early return. This isn't an error condition. It follows the same pattern as the rest of the function.

Reply:
Early returns ease readability and reduce nesting. This code incurs 3 levels of nesting that could be trivially cut out just by early returns.

Reply:
But look at the rest of the function. This is not an early return condition. I know this is the last clause in the function, but adding an early return here would just make maintenance of this function harder because it doesn't fit the general pattern.

Reply:
Left to your discretion :-)